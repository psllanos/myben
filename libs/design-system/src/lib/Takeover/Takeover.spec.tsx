import { render, screen } from '@testing-library/react'
import { Takeover } from './'

describe('Takeover', () => {
    describe('when rendered with `open` prop true', () => {
        it('should display the contents', () => {
            const onToggleMock = jest.fn()
            const component = render(
                <Takeover open={true} onClose={onToggleMock}>
                    Takeover content
                </Takeover>
            )

            expect(screen.getByText('Takeover content')).toBeInTheDocument()
            expect(component).toMatchSnapshot()
        })
    })
})
