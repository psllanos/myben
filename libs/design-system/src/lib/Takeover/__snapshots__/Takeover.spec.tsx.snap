// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Takeover when rendered with \`open\` prop true should display the contents 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
      />
    </div>
    <div
      id="headlessui-portal-root"
    >
      <div
        data-headlessui-portal=""
      >
        <button
          aria-hidden="true"
          style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px;"
          type="button"
        />
        <div>
          <div
            aria-modal="true"
            data-headlessui-state="open"
            id="headlessui-dialog-:r0:"
            role="dialog"
          >
            <div
              class="z-50 fixed inset-0 bg-black custom-gray-scroll"
            >
              <div
                class="inset-0 overflow-hidden"
                data-headlessui-state="open"
                id="headlessui-dialog-panel-:r1:"
              >
                Takeover content
              </div>
            </div>
          </div>
        </div>
        <button
          aria-hidden="true"
          style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px;"
          type="button"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
